### PR TITLE
dp/add project roles tests

### PR DESCRIPTION
**Pull request by Dpalazzari**

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -12,11 +12,12 @@ def create
     @project = Project.create_project(project_params, current_user)
     if @project.save
       @project.owners << current_user
+      current_user.add_owner_role
```
Comment (kheppenstall) [inline app/controllers/projects_controller.rb]: Maybe change this to `current_user.add_owner_role unless current_user.owner?`
@Dpalazzari

Comment (kheppenstall) [inline app/controllers/projects_controller.rb]: And add a test for this...

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -12,11 +12,12 @@ def create
...
       flash[:success] = "You successfully created #{@project.name}."
 
       redirect_to project_path(@project.slug)
     else
-      flash[:danger] = "You must fill all fields in for a project."
+      flash[:danger] = "You must fill in all fields to create a valid project."
```
Comment (kheppenstall) [inline app/controllers/projects_controller.rb]: Could refactor these to `redirect_to start_path, danger: "You must fill in all fields to create a valid project."`

@Dpalazzari

File: spec/support/feature_helpers.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 module FeatureHelpers
   def logged_in_user
     user = create(:user)
+    user.roles << create(:role)
```
Comment (kheppenstall) [inline spec/support/feature_helpers.rb]: This line shouldn't be necessary if the `before_save` on user is working correctly. Maybe pry in and check this out to see if it is redundant.
@Dpalazzari

File: spec/features/user/user_creates_new_project_spec.rb [hunk 4]
```diff
@@ -1,33 +1,43 @@
...
-  context 'fails to create project' do
-    it 'sees flash message' do
-      create_invalid_project(@category)
-      click_on "Save and continue"
+    context 'fails to create project' do
+      it 'sees flash message' do
+        create_invalid_project(@category)
+        click_on "Save and continue"
+
+        expect(@user.roles.pluck(:name)).to_not include('project_owner')
```
Comment (kheppenstall) [inline spec/features/user/user_creates_new_project_spec.rb]: This is some awesome sad path testing.

**Comment by kheppenstall** [comment]
@Dpalazzari Sad path testing looks awesome. Just left a few comments.

**Comment by Dpalazzari** [comment]
@kheppenstall Thanks. Ill get those issues taken care of.

**Comment by Dpalazzari** [comment]
screen shot 2017-02-11 at 9 58 24 am

**Comment by Dpalazzari** [comment]
@kheppenstall @Robbie-Smith @drod1000 @case-eee 

This PR is ready for final review.

**Comment by Robbie-Smith** [comment]
@Dpalazzari  Looks good!

**Pull request merged**